incubator/elasticsearch/templates/client-deployment.yaml

```diff
@@ -16,6 +16,10 @@ spec:
         app: {{ template "elasticsearch.name" . }}
         component: "{{ .Values.client.name }}"
         release: {{ .Release.Name }}
+      annotations:
```

Comment:
Can you move the `annotation` inside the if statement. That way it's not in the generated yaml if it has no content. There are a few places to make this change.